### ISSUE TITLE
mix deps.update doesn't update to the latest pre-release version

### ISSUE THREAD

**Issue by wojtekmach** [hexpm/hex#209]
I have the following setup:

```elixir
# mix.exs
defp deps do
  [
    {:ecto, "~> 2.0.0-beta"},
    {:httpoison, "~> 0.8.0"},
    {:poison, "~> 2.0"},
    {:exvcr, "~> 0.7", only: :test},
  ]
end
```

```elixir
# mix.lock
# ...
"ecto": {:hex, :ecto, "2.0.0-beta.0"},
# ...
```

Running `mix hex.outdated --pre` shows that ecto can be updated to `2.0.0-beta.2`. If I remove mix.lock and run `mix deps.update` it would correctly lock at `2.0.0-beta.2`, however `mix deps.update ecto` didn't do that for the above mentioned `mix.lock`. I'm not sure how likely it's for other people to run into it, and I'm not sure if it's a problem with hex or mix, but I was surprised that it didn't work and so I thought that I'll mention it.

**Comment by ericmj** [comment]
Can you share your mix.lock so it's easier for us to reproduce?

**Comment by ericmj** [comment]
It happens without the other dependencies, no need to share the rest of your mix.lock.

**Comment by wojtekmach** [comment]
@ericmj that fixed it, thanks for a quick turnaround!

**Comment by seantanly** [comment]
@ericmj Just to chip in, I'm encountering the same issue of mix deps.update not updating, for non-beta/pre versions too. Similarly, deleting the mix.lock file allows the update to work.

**Comment by wojtekmach** [comment]
Sean, which version of hex are you using? I do not think there was yet a release after the fix, so in interim folks might need to use master or the workaround.

**Comment by seantanly** [comment]
Here's my hex info.

```
$ mix hex.info
Hex:    0.11.1
Elixir: 1.2.3
OTP:    18.2.1

Built with: Elixir 1.2.3 and OTP 18.2.4

Registry file available (last updated: 2016-03-11 05:52:40)
Size: 1634kB (compressed 530kb)
Packages #: 1686
```

**Comment by wojtekmach** [comment]
@seantanly there's a new release that contains the fix so you should be good to go. See: https://github.com/hexpm/hex/commit/4fd74dd1cd5b166d00b0fe041756ce88f053e4c2#diff-4ac32a78649ca5bdd8e0ba38b7006a1eR14

**Comment by seantanly** [comment]
@wojtekmach Thanks! The issue is resolved with the new hex version 0.11.2. :)
